homeassistant/components/homematicip_cloud/lock.py

```diff
@@ -0,0 +1,103 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN as HMIPC_DOMAIN, HomematicipGenericEntity
+from .helpers import is_error_response
+
+ATTR_AUTO_RELOCK_DELAY = "auto_relock_delay"
+ATTR_DOOR_HANDLE_TYPE = "door_handle_type"
+ATTR_DOOR_LOCK_DIRECTION = "door_lock_direction"
+ATTR_DOOR_LOCK_NEUTRAL_POSITION = "door_lock_neutral_position"
+ATTR_DOOR_LOCK_TURNS = "door_lock_turns"
+
+DEVICE_DLD_ATTRIBUTES = {
+    "autoRelockDelay": ATTR_AUTO_RELOCK_DELAY,
+    "doorHandleType": ATTR_DOOR_HANDLE_TYPE,
+    "doorLockDirection": ATTR_DOOR_LOCK_DIRECTION,
+    "doorLockNeutralPosition": ATTR_DOOR_LOCK_NEUTRAL_POSITION,
+    "doorLockTurns": ATTR_DOOR_LOCK_TURNS,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the HomematicIP locks from a config entry."""
+    hap = hass.data[HMIPC_DOMAIN][config_entry.unique_id]
+
+    async_add_entities(
+        HomematicipDoorLockDrive(hap, device)
+        for device in hap.home.devices
+        if isinstance(device, AsyncDoorLockDrive)
+    )
+
+
+class HomematicipDoorLockDrive(HomematicipGenericEntity, LockEntity):
+    """Representation of the HomematicIP DoorLockDrive."""
+
+    _attr_supported_features = LockEntityFeature.OPEN
+
+    @property
+    def is_locked(self) -> bool | None:
+        """Return true if device is locked."""
+        return (
+            self._device.lockState == LockState.LOCKED
+            and self._device.motorState == MotorState.STOPPED
+        )
+
+    @property
+    def is_locking(self) -> bool:
+        """Return true if device is locking."""
+        return self._device.motorState == MotorState.CLOSING
+
+    @property
+    def is_unlocking(self) -> bool:
+        """Return true if device is unlocking."""
+        return self._device.motorState == MotorState.OPENING
+
+    async def async_lock(self, **kwargs: Any) -> None:
+        """Lock the device."""
+        result = await self._device.set_lock_state(LockState.LOCKED)
+        if is_error_response(result):
+            raise HomeAssistantError(
+                f"Error while locking the door lock ({self.name}): {json.dumps(result)}"
+            )
+
+    async def async_unlock(self, **kwargs: Any) -> None:
+        """Unlock the device."""
+        result = await self._device.set_lock_state(LockState.UNLOCKED)
+        if is_error_response(result):
+            raise HomeAssistantError(
+                f"Error while unlocking the door lock ({self.name}): {json.dumps(result)}"
+            )
+
+    async def async_open(self, **kwargs: Any) -> None:
+        """Open the door latch."""
+        result = await self._device.set_lock_state(LockState.OPEN)
+        if is_error_response(result):
```

Comment:
Totally up to you, but since you're replicating this same error handling logic multiple times, you could centralize it as a decorator.